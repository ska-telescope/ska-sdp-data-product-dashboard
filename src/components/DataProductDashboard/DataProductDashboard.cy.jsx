import React from 'react'
import { CssBaseline, ThemeProvider } from '@mui/material';
import { THEME_DARK, THEME_LIGHT } from '@ska-telescope/ska-gui-components';
import theme from '../../services/theme/theme';
import { DOWNLOAD_ID, PROD_1, PROD_2, TEST_DATA_FILE_1, TEXT_NO_API } from '../../utils/constants';
import DataProductDashboard from './DataProductDashboard';

const THEME = [THEME_DARK, THEME_LIGHT];

describe('<DataProductDashboard />', () => {
  for (const theTheme of THEME) {
    it('Theme ' + theTheme + ': Renders correctly when data is unavailable', () => {
      cy.mount(
        <ThemeProvider theme={theme(theTheme)}>
          <CssBaseline />
          <DataProductDashboard data-testid="DataProductDashboardId" />
        </ThemeProvider>,
      );
      cy.findByTestId("apiAvailability").contains(TEXT_NO_API).should("be.visible");
    })
  
    it('Theme ' + theTheme + ': Renders correctly when data is available', () => {
      cy.mount(
        <ThemeProvider theme={theme(theTheme)}>
          <CssBaseline />
          <DataProductDashboard data-testid="DataProductDashboardId" />
        </ThemeProvider>,
      );

      cy.findByTestId("availableData").contains(PROD_1).should("be.visible");
      cy.findByTestId("availableData").contains(PROD_2).should("be.visible");
    })    
  
    it('Theme ' + theTheme + ': Data is available for download on Data Product Dashboard', () => {
      cy.mount(
        <ThemeProvider theme={theme(theTheme)}>
          <CssBaseline />
          <DataProductDashboard data-testid="DataProductDashboardId" />
        </ThemeProvider>,
      );
      cy.findByTestId("availableData").contains("1").should("be.visible").click();
      cy.findByTestId(DOWNLOAD_ID).click()
      cy.readFile('cypress/data/' + TEST_DATA_FILE_1).should('contain', 'This is test file 1')
    })
  
    it('Theme ' + theTheme + ': Data products re-index endpoint is called', () => {
      cy.mount(
        <ThemeProvider theme={theme(theTheme)}>
          <CssBaseline />
          <DataProductDashboard data-testid="DataProductDashboardId" />
        </ThemeProvider>,
      );
      cy.findByTestId("RefreshIcon").click()
    })
  
    it('Theme ' + theTheme + ': Data products reload endpoint is called', () => {
      cy.mount(
        <ThemeProvider theme={theme(theTheme)}>
          <CssBaseline />
          <DataProductDashboard data-testid="DataProductDashboardId" />
        </ThemeProvider>,
      );
      cy.findByTestId("CachedIcon").invoke('css', 'pointer-events', 'auto')
      cy.findByTestId("CachedIcon").invoke('prop', 'disabled', false)
      cy.findByTestId("CachedIcon").click()
  });
  }
});