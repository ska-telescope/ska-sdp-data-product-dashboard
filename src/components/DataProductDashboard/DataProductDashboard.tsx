import React from 'react';
import axios from 'axios';
import { useTranslation } from 'react-i18next';

import { Box, Card, CardContent, Grid, Typography } from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import LibraryAddOutlinedIcon from '@mui/icons-material/LibraryAddOutlined';
import IndeterminateCheckBoxOutlinedIcon from '@mui/icons-material/IndeterminateCheckBoxOutlined';
import CachedIcon from '@mui/icons-material/Cached';
import RefreshIcon from '@mui/icons-material/Refresh';

import { Button, DateEntry, TextEntry, ButtonColorTypes } from '@ska-telescope/ska-gui-components';
import { ButtonVariantTypes } from '@ska-telescope/ska-gui-components';

import DataProductsTable from '@components/DataProductsTable/DataProductsTable';
import MetadataCard from '@components/MetadataCard/MetadataCard';
import { GetAPIStatus } from '@services/GetAPIStatus/GetAPIStatus';
import { API_REFRESH_RATE, SKA_DATAPRODUCT_API_URL, FILTERCARDHEIGHT } from '@utils/constants';
import DataproductDataGrid from '@components/DataGrid/DataGrid';
import DataAnnotationsCard from '@components/DataAnnotationsCard/DataAnnotationsCard';

const DataProductDashboard = () => {
  const { t } = useTranslation('dpd');
  const [updating, setUpdating] = React.useState(false);
  const [selectedFileNames, setSelectedFileNames] = React.useState({
    execution_block: '',
    relativePathName: '',
    metaDataFile: '',
    uuid: ''
  });
  const [apiRunning, updateApiRunning] = React.useState(false);
  const [apiIndexing, updateApiIndexing] = React.useState(false);
  const [newDataAvailable, updateNewDataAvailable] = React.useState(false);
  const [dataStoreLastModifiedTime, setDataStoreLastModifiedTime] = React.useState(null);
  const [initFlag, setInitFlag] = React.useState(true);

  const DEF_START_DATE = '1970-01-01';
  const DEF_END_DATE = '2070-12-31';
  const [startDate, updateStartDate] = React.useState('');
  const [endDate, updateEndDate] = React.useState('');
  const [formFields, setFormFields] = React.useState([
    { field: '', operator: 'contains', value: '' }
  ]);

  const [searchPanelOptions, setSearchPanelOptions] = React.useState({
    items: [
      {
        field: 'date_created',
        operator: 'greaterThan',
        value: DEF_START_DATE
      },
      {
        field: 'date_created',
        operator: 'lessThan',
        value: DEF_END_DATE
      }
    ],
    logicOperator: 'and'
  });

  async function CheckForNewData() {
    const results = await GetAPIStatus();
    if (results?.data) {
      updateApiRunning(results.data?.api_running ?? false);
      updateApiIndexing(results.data.metadata_store_status.indexing);
      setDataStoreLastModifiedTime(results.data.metadata_store_status?.last_metadata_update_time);
    } else {
      setDataStoreLastModifiedTime(null);
    }
  }

  async function PeriodicAPIStatusCheck() {
    React.useEffect(() => {
      CheckForNewData();
      const interval = setInterval(async () => {
        CheckForNewData();
      }, API_REFRESH_RATE);
      return () => clearInterval(interval);
    }, []);
    return;
  }

  PeriodicAPIStatusCheck();

  React.useEffect(() => {
    if (!initFlag) {
      updateNewDataAvailable(true);
    }
    if (dataStoreLastModifiedTime !== null) {
      setInitFlag(false);
    }
  }, [initFlag, dataStoreLastModifiedTime]);

  React.useEffect(() => {
    setUpdating(true);
  }, [newDataAvailable]);

  React.useEffect(() => {
    setSearchPanelOptions({
      items: [
        ...formFields,
        {
          field: 'date_created',
          operator: 'greaterThan',
          value: startDate
        },
        {
          field: 'date_created',
          operator: 'lessThan',
          value: endDate
        }
      ],
      logicOperator: 'and'
    });
  }, [startDate, endDate, formFields]);

  React.useEffect(() => {
    async function updateSearchResults() {
      setUpdating(false);
      updateNewDataAvailable(false);
    }

    if (updating) {
      updateSearchResults();
    }
  }, [endDate, formFields, startDate, updating]);

  const handleRowClick = () => {
    const selectedDataProduct = localStorage.getItem('selectedDataProduct');
    const selectedFileNames = selectedDataProduct ? JSON.parse(selectedDataProduct) : null;
    setSelectedFileNames(selectedFileNames);
  };

  async function indexDataProduct() {
    const apiUrl = SKA_DATAPRODUCT_API_URL;
    try {
      return await axios.get(`${apiUrl}/reindexdataproducts`, {
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json'
        }
      });
    } catch (e) {
      return t('error.API_NOT_AVAILABLE');
    }
  }

  async function OnClickIndexDataProduct() {
    updateApiIndexing(true);
    indexDataProduct();
    CheckForNewData();
  }

  function RenderSearchBox() {
    const handleKeyPairChange = (event: string, index: number) => {
      let data = [...formFields];
      data[index]['field'] = event;
      setFormFields(data);
    };

    const handleValuePairChange = (event: string, index: number) => {
      let data = [...formFields];
      data[index]['operator'] = 'contains';
      data[index]['value'] = event;
      setFormFields(data);
    };

    const addFields = () => {
      let object = {
        field: '',
        operator: 'contains',
        value: ''
      };

      setFormFields([...formFields, object]);
    };

    const removeFields = (index: number) => {
      let data = [...formFields];
      data.splice(index, 1);
      setFormFields(data);
    };
    return (
      <Box m={1}>
        <Card variant="outlined" sx={{ maxHeight: FILTERCARDHEIGHT, overflow: 'auto' }}>
          <CardContent>
            <Typography
              data-testid={'metaDataDescription'}
              sx={{ fontSize: 14 }}
              color="text.secondary"
              gutterBottom
            >
              {t('prompt.filter')}
            </Typography>
            <Grid
              container
              direction="row"
              spacing={1}
              justifyContent="space-between"
              alignItems="center"
            >
              <Grid item xs={12} md={6}>
                <DateEntry
                  testId="DateEntryStartdate"
                  label={t('label.startDate')}
                  setValue={(e: React.SetStateAction<string>) => updateStartDate(e)}
                  value={startDate}
                />
              </Grid>
              <Grid item xs={12} md={6}>
                <DateEntry
                  testId="DateEntryEndDate"
                  label={t('label.endDate')}
                  setValue={(e: React.SetStateAction<string>) => updateEndDate(e)}
                  value={endDate}
                />
              </Grid>

              {formFields.map((form, index) => {
                return (
                  <>
                    <Grid item xs={12}>
                      <TextEntry
                        ariaTitle="field"
                        testId="DateEntryKeyPair"
                        label={t('label.key')}
                        setValue={(event: any) => handleKeyPairChange(event, index)}
                        value={form.field}
                      />
                    </Grid>

                    <Grid item xs={12}>
                      <TextEntry
                        ariaTitle="value"
                        testId="DateEntryValuePair"
                        label={t('label.value')}
                        setValue={(event: any) => handleValuePairChange(event, index)}
                        value={form.value}
                      />
                    </Grid>

                    <Grid item xs={-4}>
                      <Button
                        testId="RemoveKeyValuePairButton"
                        color={ButtonColorTypes.Secondary}
                        disabled={updating}
                        icon={<IndeterminateCheckBoxOutlinedIcon />}
                        label="Remove"
                        onClick={() => removeFields(index)}
                        toolTip="Remove Key/Value pair"
                        variant={ButtonVariantTypes.Outlined}
                      />
                    </Grid>
                  </>
                );
              })}

              <Grid item xs={4}>
                <Button
                  testId="AddKeyValuePairButton"
                  color={ButtonColorTypes.Secondary}
                  disabled={updating}
                  icon={<LibraryAddOutlinedIcon />}
                  label="Add"
                  onClick={addFields}
                  toolTip="Add Key/Value pair"
                  variant={ButtonVariantTypes.Outlined}
                />
              </Grid>
            </Grid>

            <br />

            <Grid item xs={4}>
              <Button
                testId="SubmitSearchButton"
                color={ButtonColorTypes.Secondary}
                disabled={updating}
                icon={<SearchIcon />}
                label={t('button.search')}
                onClick={() => setUpdating(true)}
                toolTip="Submit search"
                variant={ButtonVariantTypes.Outlined}
              />
            </Grid>
          </CardContent>
        </Card>
      </Box>
    );
  }

  function RenderDataStoreBox() {
    return (
      <Box m={1}>
        <Grid container spacing={1} direction="row" justifyContent="justify-left">
          <Grid item>
            <Button
              testId="IndexDataProductsButton"
              color={ButtonColorTypes.Secondary}
              disabled={apiIndexing}
              icon={<RefreshIcon />}
              label={t('button.indexDP')}
              onClick={() => OnClickIndexDataProduct()}
              toolTip="Re-index all the data product files found on the storage volume. The resulting index is updated in the memory of the backend API."
              variant={ButtonVariantTypes.Outlined}
            />
          </Grid>
          <Grid item>
            <Button
              testId="ReloadDataProductsButton"
              color={ButtonColorTypes.Secondary}
              disabled={!newDataAvailable}
              icon={<CachedIcon />}
              label={t('button.reload')}
              onClick={() => setUpdating(true)}
              toolTip="Load the latest index from the backend API, update the index in the browser and reloads the data product table."
              variant={ButtonVariantTypes.Outlined}
            />
          </Grid>
        </Grid>
      </Box>
    );
  }

  return (
    <Box sx={{ height: '100%' }}>
      {RenderDataStoreBox()}
      <Grid
        sx={{ height: '100%' }}
        container
        spacing={1}
        direction="row"
        justifyContent="space-between"
      >
        <Grid item xs={9}>
          {DataProductsTable(
            updating,
            apiRunning,
            DataproductDataGrid(handleRowClick, searchPanelOptions, updating)
          )}
        </Grid>
        <Grid item xs={3}>
          <>
            {RenderSearchBox()}
            {MetadataCard(selectedFileNames)}
            {DataAnnotationsCard(selectedFileNames)}
          </>
        </Grid>
      </Grid>
    </Box>
  );
};

export default DataProductDashboard;
