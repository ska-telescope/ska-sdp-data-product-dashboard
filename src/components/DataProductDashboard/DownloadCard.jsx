import * as React from 'react';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import DownloadIcon from '@mui/icons-material/Download';
import { onDownloadClick } from '../../services/DataProduct/DataProductDownload';

// Moved the onDownloadClick to DataProductDownload as it is easier to test. 


export default function DownloadCard(selectedFileNames) {
  const { fileName } = selectedFileNames
  return (
    <Card sx={{ minWidth: 275 }}>
      <CardContent>
        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
          Selected file:
        </Typography>
        <Typography variant="h5" component="div">
          {fileName} 
        </Typography>
      </CardContent>
      <CardActions>
        <Button variant="outlined" color="secondary" onClick={() => onDownloadClick(selectedFileNames)}>
          <DownloadIcon />
          Download
        </Button>
      </CardActions>
    </Card>
  );
}
