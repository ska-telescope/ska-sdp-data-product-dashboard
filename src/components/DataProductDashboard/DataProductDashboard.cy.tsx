import React from 'react';
import { CssBaseline, ThemeProvider } from '@mui/material';
import { THEME_DARK, THEME_LIGHT } from '@ska-telescope/ska-gui-components';
import theme from '@services/theme/theme';
import { DOWNLOAD_ID, PROD_1, PROD_2, TEST_DATA_FILE_1, TEXT_NO_API } from '@utils/constants';
import DataProductDashboard from '@components/DataProductDashboard/DataProductDashboard';
import MockStatusAPINotRunning from '@services/Mocking/mockStatusAPINotRunning';
import { StoreProvider } from '@ska-telescope/ska-gui-local-storage';
import AuthProvider from '@components/Auth/AuthContext';

const THEME = [THEME_DARK, THEME_LIGHT];

describe('<DataProductDashboard />', () => {
  for (const theTheme of THEME) {
    it('Theme ' + theTheme + ': Renders correctly when data is unavailable', () => {
      const stub = cy.stub().resolves(MockStatusAPINotRunning);
      cy.window().then((win) => (win.GetAPIStatus = stub));
      cy.mount(
        <StoreProvider>
          <AuthProvider>
            <React.StrictMode>
              <ThemeProvider theme={theme(theTheme)}>
                <CssBaseline />
                <DataProductDashboard data-testid="DataProductDashboardId" />
              </ThemeProvider>
            </React.StrictMode>
          </AuthProvider>
        </StoreProvider>
      );
      cy.findByTestId('apiAvailability').contains(TEXT_NO_API).should('be.visible');
    });

    it('Theme ' + theTheme + ': Renders correctly when data is available', () => {
      cy.mount(
        <StoreProvider>
          <AuthProvider>
            <React.StrictMode>
              <ThemeProvider theme={theme(theTheme)}>
                <CssBaseline />
                <DataProductDashboard data-testid="DataProductDashboardId" />
              </ThemeProvider>
            </React.StrictMode>
          </AuthProvider>
        </StoreProvider>
      );

      cy.findByTestId('availableData').contains(PROD_1).should('be.visible');
      cy.findByTestId('availableData').contains(PROD_2).should('be.visible');
    });

    it('Theme ' + theTheme + ': Data is available for download on Data Product Dashboard', () => {
      cy.mount(
        <StoreProvider>
          <AuthProvider>
            <React.StrictMode>
              <ThemeProvider theme={theme(theTheme)}>
                <CssBaseline />
                <DataProductDashboard data-testid="DataProductDashboardId" />
              </ThemeProvider>
            </React.StrictMode>
          </AuthProvider>
        </StoreProvider>
      );
      cy.findByTestId('availableData').contains(PROD_1).click();
      cy.findByTestId(DOWNLOAD_ID).click();
      cy.readFile('cypress/data/' + TEST_DATA_FILE_1).should('contain', 'This is test file 1');
    });

    it('Theme ' + theTheme + ': Data products re-index endpoint is called', () => {
      cy.mount(
        <StoreProvider>
          <AuthProvider>
            <React.StrictMode>
              <ThemeProvider theme={theme(theTheme)}>
                <CssBaseline />
                <DataProductDashboard data-testid="DataProductDashboardId" />
              </ThemeProvider>
            </React.StrictMode>
          </AuthProvider>
        </StoreProvider>
      );
      cy.findByTestId('RefreshIcon').click();
    });

    it('Theme ' + theTheme + ': Data products reload endpoint is called', () => {
      cy.mount(
        <StoreProvider>
          <AuthProvider>
            <React.StrictMode>
              <ThemeProvider theme={theme(theTheme)}>
                <CssBaseline />
                <DataProductDashboard data-testid="DataProductDashboardId" />
              </ThemeProvider>
            </React.StrictMode>
          </AuthProvider>
        </StoreProvider>
      );
      cy.findByTestId('CachedIcon').invoke('css', 'pointer-events', 'auto');
      cy.findByTestId('CachedIcon').invoke('prop', 'disabled', false);
      cy.findByTestId('CachedIcon').click();
    });
  }
});
