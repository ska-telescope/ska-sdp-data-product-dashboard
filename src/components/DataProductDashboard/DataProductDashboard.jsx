import React from 'react';
import { Grid, Typography } from '@mui/material';
import TreeView from '@mui/lab/TreeView';
import TreeItem from '@mui/lab/TreeItem';
import DownloadCard from '../DownloadCard/DownloadCard';
import MetaDataComponent from '../MetaDataComponent/MetaDataComponent';
import DataProductList from '../../services/DataProduct/DataProductList';
import MetaData from '../../services/MetaData/MetaData';
import WarningIcon from '@mui/icons-material/Warning';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';

const TREE_MAX_WIDTH = 500;
const TREE_HEIGHT = 500;

const DataProductDashboard = () => {
  const [jsonDataProductsTree, setJsonDataProductsTree] = React.useState({data:[]});
  const [metaData, setMetaData] = React.useState({data:[]});
  const [oldFilename, setOldFilename] = React.useState(null);
  const [selectedFileNames, setSelectedFileNames] = React.useState({
    fileName: '',
    relativeFileName: '',
    metaDataFile: ''
  });

  React.useEffect(() => {
    setOldFilename(selectedFileNames.metaDataFile);
  }, [metaData]);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const getSelectedNodeInfo = (jsonTree, nodeId) => {
    // Test if the jsonTree is defined.
    if (typeof jsonTree !== 'undefined') {
      // Evaluate first elements of the jsonTree, it is of type array, reevaluate each element of the array.
      if (jsonTree instanceof Array) {
        for (let i = 0; i < jsonTree.length + 1; i += 1) {
          getSelectedNodeInfo(jsonTree[i], nodeId);
        }
      } else {
        // If the element of the array is not an array, evaluate its properties. If matching ID is found, update the setSelectedFileNames.
        Object.keys(jsonTree).forEach(prop => {
          if (jsonTree.id.toString() === nodeId) {
            setSelectedFileNames({
              fileName: jsonTree.name,
              relativeFileName: jsonTree.relativefilename,
              metaDataFile: jsonTree.metadatafile
            });
            return;
          }
          if (jsonTree[prop] instanceof Object || jsonTree[prop] instanceof Array) {
            getSelectedNodeInfo(jsonTree[prop], nodeId);
          }
        });
      }
    }
  };

  const handleSelectedNode = (_event, nodeId) => {
    getSelectedNodeInfo(jsonDataProductsTree.data, nodeId);
  };

  async function getJsonDataProductsTree() {
    setJsonDataProductsTree(await DataProductList());
  }

  async function getMetaData() {
    const results = await MetaData(selectedFileNames?.metaDataFile);
    setMetaData(results.data);
  }

  if (jsonDataProductsTree.status === undefined) {
    getJsonDataProductsTree();
  }

  function renderDataProductsTreeNodes() {
    if (jsonDataProductsTree.status === 200) {
      const renderTree = nodes => (
        <TreeItem key={nodes.id} nodeId={nodes.id.toString()} label={nodes.name}>
          {Array.isArray(nodes.children) ? nodes.children.map(node => renderTree(node)) : null}
        </TreeItem>
      );
      return renderTree(jsonDataProductsTree.data);
    }
    return <></>;
  }

  function renderDataProductsTreeComponent() {
    if (jsonDataProductsTree.status === 200 ) {
      return (
        <TreeView
          aria-label="rich object"
          defaultCollapseIcon={<ExpandMoreIcon />}       
          defaultExpanded={['root']}
          defaultExpandIcon={<ChevronRightIcon />}
          onNodeSelect={handleSelectedNode}
          sx={{ height: TREE_HEIGHT, flexGrow: 1, maxWidth: TREE_MAX_WIDTH, overflowY: 'auto' }}
        >
          {jsonDataProductsTree.data && renderDataProductsTreeNodes()}
        </TreeView>
      );
    }
    return (
      <>
        <Typography sx={{ fontSize: 25, display: "flex", justifyContent: "center" }} color="#D33115" gutterBottom>
          <WarningIcon sx={{ fontSize: "35px" }}  />
          {" "}
          SDP Data API not available
        </Typography>
      </>
    );
  }

  function RenderDownloadCard() {
    if ( selectedFileNames.relativeFileName !== '' ) {
      return (
       DownloadCard(selectedFileNames)
      );
    }
  }

  const displayData = () => {
    let result = false;
    const metaDataFile = selectedFileNames?.metaDataFile;

    if (metaDataFile && metaDataFile.length) {
      result = true;
      if (oldFilename !== metaDataFile) {
        getMetaData();
      }
    }
    return result;
  }

  return (
    <>
      <Grid container direction="row" justifyContent="space-between">
        <Grid item xs={6}>
          {renderDataProductsTreeComponent()}
        </Grid>
        <Grid item xs={6}>
          <>
            {RenderDownloadCard()}
            {displayData() && MetaDataComponent(metaData)}
          </>
        </Grid>
      </Grid>
    </>
  );
};

export default DataProductDashboard;
