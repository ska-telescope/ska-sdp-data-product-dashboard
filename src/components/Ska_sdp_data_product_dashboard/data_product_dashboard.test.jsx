import React from 'react';
import { shallow } from 'enzyme';
import { cleanup } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import DataProductDashboard from './data_product_dashboard';

process.env.REACT_APP_SKA_SDP_DATA_PRODUCT_DUMMY_DATA = 'true';

// afterEach function runs after each test suite is executed
afterEach(() => {
  cleanup(); // Resets the DOM after each test suite
});

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  // restore the spy created with spyOn
  jest.restoreAllMocks();
});

describe('Data Product Dashboard', () => {
  it('renders', () => {
    const wrapper = shallow(<DataProductDashboard />);
    expect(wrapper.exists());
  });

  describe('TreeView', () => {
    it('renders', () => {
      const wrapper = shallow(<DataProductDashboard />);
      expect(wrapper.find('TreeView').exists());
    });
  });

  describe('Download Button', () => {
    it('renders', () => {
      const wrapper = shallow(<DataProductDashboard />);
      expect(wrapper.find('button').exists());
      expect(wrapper.find('button').contains('DOWNLOAD'));
    });
  });
});
