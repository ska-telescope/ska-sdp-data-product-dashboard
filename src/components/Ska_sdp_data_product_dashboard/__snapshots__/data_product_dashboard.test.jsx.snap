// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user data 1`] = `
<Fragment>
  <ForwardRef(TreeView)
    aria-label="rich object"
    defaultCollapseIcon={<Memo />}
    defaultExpandIcon={<Memo />}
    defaultExpanded={
      Array [
        "root",
      ]
    }
    onNodeSelect={[Function]}
    sx={
      Object {
        "flexGrow": 1,
        "height": 500,
        "maxWidth": 500,
        "overflowY": "auto",
      }
    }
  />
  <ForwardRef(Button)
    color="secondary"
    onClick={[Function]}
    variant="outlined"
  >
    <Memo(ForwardRef(DownloadIcon)) />
    Download
  </ForwardRef(Button)>
</Fragment>
`;
